Java/Rathna/IsValidParenthesis.java

```diff
@@ -7,8 +8,34 @@ public static void main(String[] args) {
         String inputStr = scan.next();
         System.out.println("Contents of input string: " + inputStr);
         IsValidParenthesis(inputStr);
+        boolean bool = IsValidParenthesisUsingStack(inputStr);
+        if(bool) {
+            System.out.println("Good Input");
+        } else {
+            System.out.println("Bad Input");
+        }
         scan.close();
     }
+
+    private static boolean IsValidParenthesisUsingStack(String inputStr) {
+        Stack<Character> stack = new Stack<>();
+        for(int i = 0; i < inputStr.length(); i++) {
+            if(inputStr.charAt(i) == '(' || inputStr.charAt(i) == '{' || inputStr.charAt(i) == '[') {
+                stack.push(inputStr.charAt(i)); // SC = O(n), TC = (n)
+            }
+            else {
+                char poppedValue = stack.pop();
+                if((inputStr.charAt(i) == ')' && poppedValue != '(') ||
```

Comment:
combining popped value and nextChar of string and then comparing with ()/{}/[] is failing as String comparison is not same as content comparison, so changed to normal way.